seam/seam.py

```diff
@@ -1,51 +1,58 @@
 import os
-
-from .routes import Routes
 import requests
 from importlib.metadata import version
 from typing import Optional, Union, Dict, cast
+from typing_extensions import Self
+from seam.utils.auth import get_auth_headers
+from seam.utils.parse_options import get_api_key_from_env, get_endpoint_from_env
+from .routes import Routes
 from .types import AbstractSeam, SeamApiException
 
 
+DEFAULT_ENDPOINT = "https://connect.getseam.com"
+
+
 class Seam(AbstractSeam):
     """
     Initial Seam class used to interact with Seam API
     """
 
-    api_key: str
-    api_url: str = "https://connect.getseam.com"
     lts_version: str = "1.0.0"
 
     def __init__(
         self,
         api_key: Optional[str] = None,
         *,
+        personal_access_token: Optional[str] = None,
         workspace_id: Optional[str] = None,
-        api_url: Optional[str] = None,
+        endpoint: Optional[str] = None,
         wait_for_action_attempt: Optional[Union[bool, Dict[str, float]]] = False,
     ):
         """
         Parameters
         ----------
         api_key : str, optional
-          API key
+          API key.
+        personal_access_token : str, optional
+          Personal access token.
         workspace_id : str, optional
-          Workspace id
-        api_url : str, optional
-          API url
+          Workspace id.
+        endpoint : str, optional
+          The API endpoint to which the request should be sent.
+        wait_for_action_attempt : bool or dict, optional
+          Controls whether to wait for an action attempt to complete, either as a boolean or as a dictionary specifying `timeout` and `poll_interval`. Defaults to `False`.
         """
+
         Routes.__init__(self)
 
-        if api_key is None:
-            api_key = os.environ.get("SEAM_API_KEY", None)
-        if api_key is None:
-            raise Exception(
-                "SEAM_API_KEY not found in environment, and api_key not provided"
-            )
-        if workspace_id is None:
-            workspace_id = os.environ.get("SEAM_WORKSPACE_ID", None)
-        self.api_key = api_key
-        self.workspace_id = workspace_id
+        api_key = api_key or get_api_key_from_env(
+            personal_access_token=personal_access_token,
```

Comment:
This should not depend on personal_access_token. Better to add a function to `parse_options.py` that returns all of the options and throws SeamHttpInvalidOptionsError when needed.

Reply:
Got it. I was trying to introduce similar logic to what we have on JS SDK [here](https://github.com/seamapi/javascript-http/blob/ca078db6f0231bdaef7ac3f92693ad01d776cb60/src/lib/seam/connect/parse-options.ts#L86-L99). Don't read api key from env if other auth args were passed, in python's case it only relates to PAT.

Reply:
Is this better 4eb625cfa4ae89395c48544bc97d50f95610671e?

Reply:
> don't read api key from env if other auth args are passed, in python's case it only relates to PAT.
Yes, this is the right logic :)